ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerPolicy.java

```diff
@@ -0,0 +1,1513 @@
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
+      toString(sb);
+      return sb.toString();
+    }
+
+    public StringBuilder toString(StringBuilder sb) {
+      sb.append("RangerDataMaskPolicyItem={");
+      super.toString(sb);
+      sb.append("dataMaskInfo={");
+      if (dataMaskInfo != null) {
+        dataMaskInfo.toString(sb);
+      }
+      sb.append("} ");
+      sb.append("}");
+      return sb;
+    }
+  }
+
+  /**
+   * RangerRowFilterPolicyItem class.
+   */
+  @JsonAutoDetect(fieldVisibility = Visibility.ANY)
+  @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  @XmlRootElement
+  @XmlAccessorType(XmlAccessType.FIELD)
+  public static class RangerRowFilterPolicyItem extends RangerPolicyItem implements java.io.Serializable {
+    private static final long serialVersionUID = 1L;
+    private RangerPolicyItemRowFilterInfo rowFilterInfo;
+
+    public RangerRowFilterPolicyItem() {
+      this(null, null, null, null, null, null);
+    }
+
+    public RangerRowFilterPolicyItem(RangerPolicyItemRowFilterInfo rowFilterInfo,
+                                     List<RangerPolicyItemAccess> accesses, List<String> users, List<String> groups,
+                                     List<RangerPolicyItemCondition> conditions, Boolean delegateAdmin) {
+      super(accesses, users, groups, conditions, delegateAdmin);
+      setRowFilterInfo(rowFilterInfo);
+    }
+
+    /**
+     * @return the rowFilterInfo
+     */
+    public RangerPolicyItemRowFilterInfo getRowFilterInfo() {
+      return rowFilterInfo;
+    }
+
+    /**
+     * @param rowFilterInfo the rowFilterInfo to set
+     */
+    public void setRowFilterInfo(RangerPolicyItemRowFilterInfo rowFilterInfo) {
+      this.rowFilterInfo = rowFilterInfo == null ? new RangerPolicyItemRowFilterInfo() : rowFilterInfo;
+    }
+
+    @Override
+    public int hashCode() {
+      final int prime = 31;
+      int result = super.hashCode();
+      result = prime * result + ((rowFilterInfo == null) ? 0 : rowFilterInfo.hashCode());
+      return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (!super.equals(obj)) {
+        return false;
+      }
+      if (this == obj) {
+        return true;
+      }
+      if (obj == null) {
+        return false;
+      }
+      if (getClass() != obj.getClass()) {
+        return false;
+      }
+      RangerRowFilterPolicyItem other = (RangerRowFilterPolicyItem) obj;
+      if (rowFilterInfo == null) {
+        if (other.rowFilterInfo != null) {
+          return false;
+        }
+      } else if (!rowFilterInfo.equals(other.rowFilterInfo)) {
+        return false;
+      }
+      return true;
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
+      toString(sb);
+      return sb.toString();
+    }
+
+    public StringBuilder toString(StringBuilder sb) {
```

Comment:
Mark it private, may change the signature as well. It is toString() but returns StringBuilder